src/index.ts

```diff
@@ -0,0 +1,76 @@
+import * as core from "@actions/core";
+import { exec } from "@actions/exec";
+import { hasWiki } from "./has-wiki";
+import { isDirty } from "./is-dirty";
+
+// TODO: Create (push) commit via the API?
+//      https://docs.github.com/en/rest/git/commits?apiVersion=2022-11-28#create-a-commit
+async function run() {
+  const token = core.getInput("token", { required: true });
+  const repository = core.getInput("repository", { required: true });
+  const path = core.getInput("path", { required: true });
+  const commitMessage = core.getInput("commit-message", { required: true });
+  const dryRun = core.getBooleanInput("dry-run", { required: true });
+  const user = process.env["GITHUB_ACTOR"] || "github-actions[bot]";
+  const wikiRemote = `https://${user}:${token}@${repository}.wiki.git`;
+  if (!(await hasWiki(wikiRemote))) {
+    core.notice("You must enable the wiki and add an initial page.");
+    core.setFailed("Wiki repository not found");
+    return;
+  }
+
+  const cwd = path;
+  const baseOptions = { cwd };
+  await exec("git", ["init"], baseOptions);
```

Comment:
Remember! This still has the issue with the existing `.git` folder from the _main repo_ that was `actions/checkout`-ed into here! [#72](https://github.com/spenserblack/actions-wiki/issues/72)

Reply:
Yup! This will be addressed in a subsequent change.